jasmine/spec/inverted-index-test.js

```diff
@@ -18,36 +18,30 @@ describe('Read book data', () => {
   });
   it('parses received files into JSON objects', () => {
     expect(typeof (invertedIndexTest.validFiles(wrongFile))).toBe(typeof {});
-  });
-  it('reads a JSON file and asserts that it valid', () => {
-    expect(invertedIndexTest.validFiles(JSON.stringify(validBook))[1]).toBe('Success');
-  });
-  it('reads a JSON file and asserts that it valid', () => {
-    expect(invertedIndexTest.validFiles(JSON.stringify(trollTest))[1]).toBe('this Index takes books with Title and Text property only');
-  });
+  }); 
   it('reads a JSON file and asserts that it valid', () => {
-    expect(invertedIndexTest.validFiles(JSON.stringify(wrongFile))[1]).toBe('this Index takes books with Title and Text property only');
+     expect(invertedIndexTest.validFiles(JSON.stringify(validBook))[1]).toBe('Success');
   });
 });
 
 describe('Populate Index', () => {
   it('verifies that an Index is created once the JSON file has been read', () => {
-    expect(invertedIndexTest.createIndex('booky.json', validBook)['booky.json'].alice).toBeDefined();
+    expect(invertedIndexTest.createIndex(validBook).alice).toBeDefined();
   });
   it('should map the string keys to the correct objects in the JSON array', () => {
-    expect(invertedIndexTest.createIndex('books.json', validBook)['books.json'].of).toEqual({ 0: true, 1: true });
+    expect(invertedIndexTest.createIndex(validBook).of).toEqual({ 0: true, 1: true });
   });
   it('Should confirm correct index is returned', () => {
-    expect(invertedIndexTest.getIndex()['booky.json'].of).toEqual({ 0: true, 1: true });
-    expect(Object.keys(invertedIndexTest.getIndex()['booky.json']).length).toEqual(25);
+    expect(invertedIndexTest.getIndex().of).toEqual({ 0: true, 1: true });
+    expect(Object.keys(invertedIndexTest.getIndex()).length).toEqual(25);
   });
 });
 
 describe('Search index', () => {
   it('returns an array of the indices of the correct objects that contain the words in the search query', () => {
-    expect(invertedIndexTest.searchIndex('booky.json', 'Alice Bicycle')['booky.json'].alice).toEqual({ 0: true });
+    expect(invertedIndexTest.searchIndex('Alice Bicycle').alice).toEqual({ 0: true });
```

Comment:
###Nitpick

Test for a case where it returns the exact search result for the searched files